vchan/io.c

```diff
@@ -62,22 +62,33 @@ int libvchan__check_domain_alive(xc_interface *xc_handle, int dom) {
 }
 
 int libvchan_write(libvchan_t *ctrl, const void *data, size_t size) {
+    if (!ctrl->xenvchan)
+        return -1;
     return libxenvchan_write(ctrl->xenvchan, (char*)data, size);
 }
 
 int libvchan_send(libvchan_t *ctrl, const void *data, size_t size) {
+    if (!ctrl->xenvchan)
+        return -1;
     return libxenvchan_send(ctrl->xenvchan, (char*)data, size);
 }
 
 int libvchan_read(libvchan_t *ctrl, void *data, size_t size) {
+    if (!ctrl->xenvchan)
+        return -1;
```

Comment:
Should `errno` be set to `EAGAIN`?  Same for other similar APIs below.

Reply:
Not sure if that's the best match (just retrying doesn't make sense without successful _finish call first), but also I don't have a better idea.